pandasdmx/writer.py

```diff
@@ -56,11 +52,23 @@ def write(obj, *args, **kwargs):
 
 # Functions for Python containers
 def write_list(obj, *args, **kwargs):
-    """List of objects."""
+    """Convert a :class:`list` of SDMX objects.
+
+    For the following *obj*, :meth:`write_list` returns :class:`pandas.Series`
+    instead of a :class:`list`:
+
+    - a list of :class:`Observation <pandasdmx.model.Observation>`:
+      the Observations are written using :meth:`write_dataset`.
+    - a list with only 1 :class:`DataSet <pandasdmx.model.DataSet>` (e.g. the
+       :attr:`data <pandasdmx.message.DataMessage.data>` attribute of
+       :class:`DataMessage <pandasdmx.message.DataMessage>`): the Series for
+       the single element is returned.
+    - a list of :class:`SeriesKey`: the key values (but no data) are returned.
+    """
     if isinstance(obj[0], Observation):
         return write_dataset(obj, *args, **kwargs)
-    elif isinstance(obj[0], DataSet):
-        return write_dataset(obj[0], *args, **kwargs)
+    elif isinstance(obj[0], DataSet) and len(obj) == 1:
```

Comment:
@dr-leo note this additional condition. Otherwise, in a DataMessage with 2 or more DataSets, the code would return only the first and silently discard the rest, which IMO is undesirable.

Reply:
Agreed, even if I cannot think of a DataMessage with more than one datasets.